src/mono/mono/mini/interp/transform.c

```diff
@@ -8585,6 +8585,8 @@ interp_mark_ref_slots_for_var (TransformData *td, int var)
 	if (!td->ref_slots || max_index >= td->ref_slots->size) {
 		guint32 old_size = td->ref_slots ? (guint32)td->ref_slots->size : 0;
 		guint32 new_size = old_size ? old_size * 2 : 32;
+		while (new_size <= max_index)
+			new_size *= 2;
```

Comment:
Does it matter if it allocates new_size > max_index? Should it go up to max_index instead?

Reply:
That might produce further reallocation, but it is also valid. I don't think it matters much